docs/usage.rst

```diff
@@ -1181,3 +1181,32 @@ data during a calculation, and then re-applied when returning data from a
 function. In other words, apply or check units at interfaces, but during an
 internal calculation it is often worth stripping units, especially if the
 calculation involves many operations on arrays with only a few elements.
+
+Plotting with Matplotlib
+++++++++++++++++++++++++
+
+Matplotlib is Unyt aware. With no additional effort, Matplotlib will label the x and y
+axes with the units.
+
+  >>> import matplotlib.pyplot as plt # doctest: +SKIP
+  >>> from unyt import s, K
+  >>> x = [0.0, 0.01, 0.02]*s
+  >>> y = [298.15, 308.15, 318.15]*K
+  >>> plt.plot(x, y) # doctest: +SKIP
+  [<matplotlib.lines.Line2D object at ...>] # doctest: +SKIP
+  >>> plt.show() # doctest: +SKIP
+
+.. image:: _static/mpl_fig1.png
+
+You can change the plotted units without affecting the original data.
+
+  >>> plt.plot(x, y, xunits="ms", yunits=("J", "thermal")) # doctest: +SKIP
+  [<matplotlib.lines.Line2D object at ...>] # doctest: +SKIP
+  >>> plt.show() # doctest: +SKIP
+
+.. image:: _static/mpl_fig2.png
+
+.. note::
+  
+  - This feature works in matplotlib versions 3.1 and above
```

Comment:
agree, but there is a bug in version 3.0.3 where the unyt_arrayConverter is never called, ever. It appears that since unyt_array is a subclass of ndarray, they never lookup unyt_array in their registry and instead treat it as an ndarray. Version 3.1 and forward works correctly.

Reply:
I still think we can relax the strict version requirement, especially if it's only in that one bugfix version where there's an issue. Yyou can also blacklist matplotlib 3.0.3 in tox and at import time for matplotlib by checking `matplotlib.__version__`.

Reply:
Ah, you're saying that they ignore `ndarray` subclasses for all older versions?

Reply:
yes, that appears to be the issue. here's an example when using matplotlib 3.0.3:
```python
>>> import matplotlib.pyplot as plt
>>> from unyt import *
>>> x = [0,1,2]*s
>>> y = [3,4,5]*V
>>> ax = plt.figure().add_subplot()
>>> ax.xaxis.get_label()
""
```
The label is an empty string (the default value) when it should be the expected unit label.

Reply:
OK, I just tested on a few matplotlib versions and indeed it looks like matplotlib 3.1 or newer is needed for some reason.